src/Npgsql/PGUtil.cs

```diff
@@ -138,6 +138,17 @@ internal unsafe static double ReverseEndianness(double value)
         internal static readonly Task<bool> TrueTask = Task.FromResult(true);
         internal static readonly Task<bool> FalseTask = Task.FromResult(false);
 
+        internal static Task TaskFromException(Exception exception)
+        {
+#if NET45 || NET451
+            var tcs = new TaskCompletionSource<bool>();
+            tcs.SetException(exception);
+            return tcs.Task;
+#else
+            return Task.FromException(exception);
+#endif
+        }
+
```

Comment:
Even after reading the docs I don't understand what the advantage of `Task.FromException` is over just throwing the exception from the async method.
Is this a performance issue?

Reply:
The reason to use `FromException` is to produce a failed task for which you can run continuations. It's useful when an exception isn't related to an argument validation.

Reply:
Right, but as I wrote above, the usage here actually seems pretty much like argument validation...

Reply:
Yeah, will do (: